src/connector/src/sink/starrocks.rs

```diff
@@ -583,42 +549,41 @@ impl SinkWriter for StarrocksSinkWriter {
         }
     }
 
-    async fn barrier(&mut self, is_checkpoint: bool) -> Result<Option<SinkMetadata>> {
-        if self.client.is_some() {
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<()> {
+        if let Some(client) = self.client.take() {
             // Here we finish the `/api/transaction/load` request when a barrier is received. Therefore,
             // one or more load requests should be made within one commit_checkpoint_interval period.
             // StarRocks will take care of merging those splits into a larger one during prepare transaction.
             // Thus, only one version will be produced when the transaction is committed. See Stream Load
             // transaction interface for more information.
-            let client = self
-                .client
-                .take()
-                .ok_or_else(|| SinkError::Starrocks("Can't find starrocks inserter".to_owned()))?;
             client.finish().await?;
         }
 
-        if is_checkpoint {
-            if self.curr_txn_label.is_some() {
-                let txn_label = self.curr_txn_label.take().unwrap();
-                tracing::debug!(?txn_label, "prepare transaction");
-                let txn_label_res = self.txn_client.prepare(txn_label.clone()).await?;
-                assert_eq!(
-                    txn_label, txn_label_res,
-                    "label responding from StarRocks differs from the current one"
-                );
-                Ok(Some(StarrocksWriteResult(Some(txn_label)).try_into()?))
-            } else {
-                // no data was written within previous epoch
-                Ok(Some(StarrocksWriteResult(None).try_into()?))
+        if is_checkpoint && let Some(txn_label) = self.curr_txn_label.take() {
+            if let Err(err) = self.prepare_and_commit(txn_label.clone()).await {
+                self.txn_client
+                    .rollback(txn_label.clone())
+                    .await
+                    .map_err(|err| {
```

Comment:
On rollback error, we may only add a warning log, and otherwise we will lose the information in the error of `prepare_and_commit`.